filemanager/gcsmanager.go

```diff
@@ -85,18 +101,31 @@ func (m *gcsManager) Upload(ctx context.Context, file *os.File, prefixes ...stri
 	ctx, cancel := context.WithTimeout(ctx, m.getTimeout())
 	defer cancel()
 
-	obj := client.Bucket(m.config.Bucket).Object(fileName)
-	w := obj.NewWriter(ctx)
-	if _, err := io.Copy(w, file); err != nil {
-		err = fmt.Errorf("copying file to GCS: %v", err)
-		if closeErr := w.Close(); closeErr != nil {
-			return UploadedFile{}, fmt.Errorf("closing writer: %q, while: %w", closeErr, err)
-		}
+	object := client.Bucket(m.config.Bucket).Object(fileName)
+	if m.config.doNotOverWrite {
+		object = object.If(storage.Conditions{DoesNotExist: true})
+	}
+	w := object.NewWriter(ctx)
```

Comment:
do we need to close the writer as well?
```suggestion
	w := object.NewWriter(ctx)
	defer func(){
	  _  :=  w.Close()
	}()
```

Reply:
from `func (o *storage.ObjectHandle) NewWriter(ctx context.Context) *storage.Writer` documentation, cancelling the context should be enough:
```
It is the caller's responsibility to call Close when writing is done. To stop writing without saving the data, cancel the context.
```